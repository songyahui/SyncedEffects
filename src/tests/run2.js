"use hopscript"

var hh = require("hiphop");

var m1 = <hh.module T W V Z>
     <hh.parallel>
       <hh.if T>
	 <hh.local L>
	   <hh.sequence>
	     <hh.emit L/>
	     <hh.emit V/>
	   </hh.sequence>
	 </hh.local>
       </hh.if>
       <hh.if W>
	 <hh.emit Z/>
       </hh.if>
     </hh.parallel>
   </hh.module>;

var inSig={accessibility: hh.IN};
var m2 = <hh.module S=${inSig} U=${inSig} A B>
  <hh.sequence>
    <hh.local L>
      <hh.emit L/>
    </hh.local>
    <hh.run module=${m1} T=S W=U V=A Z=B/>
    <hh.run module=${m1} T=S W=U V=A Z=B/>
  </hh.sequence>
</hh.module>;

exports.prg = new hh.ReactiveMachine(m2, "run22");
