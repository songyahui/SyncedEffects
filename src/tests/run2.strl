module m1:
input T, W;
output V, Z;
       present T then
       	       signal L in
	       	      emit L;
	       	      emit V
	       end signal
       end present
||
	present W then emit Z end
end module

module run22:
input S, U;
output A, B;
signal L in
       emit L;
end signal;
run m1 [ signal S / T,
       	 	U / W,
		A / V,
		B / Z ];
run m1 [ signal S / T,
       	 	U / W,
		A / V,
		B / Z ];
end module
