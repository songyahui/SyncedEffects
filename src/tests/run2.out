run22> S U;
--- Output: A B
run22> ;
--- Output:
